File: baselibrary/src/main/java/com/ju/baselibrary/base/BaseFragment.java
package com.ju.baselibrary.base;


import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.LayoutRes;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.ju.baselibrary.widget.gloading.Gloading;

public abstract class BaseFragment extends Fragment {
    protected Gloading.Holder mLoadHolder;
    protected Context mContext;

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        mContext = context;
    }

    protected abstract void init();

    protected abstract void initView(View view);

    protected abstract void initData();

    /**
     * 加载布局
     */
    @LayoutRes
    protected abstract int getLayoutId();

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, Bundle savedInstanceState) {
        return inflater.inflate(getLayoutId(), null);
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        init();
        initView(view);
        initData();
    }
}

File: baselib/src/main/java/com/hzy/baselib/base/BaseFragment.kt
package com.hzy.baselib.base


import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import androidx.annotation.LayoutRes
import androidx.fragment.app.Fragment

import com.hzy.baselib.widget.gloading.Gloading


abstract class BaseFragment : Fragment() {
    protected var mLoadHolder: Gloading.Holder? = null
    protected var mContext: Context? = null

    /**
     * 加载布局
     */
    @get:LayoutRes
    protected abstract val layoutId: Int

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        mContext = context
    }

    protected abstract fun init()

    protected abstract fun initView(view: View)

    protected abstract fun initData()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(layoutId, null)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        init()
        initView(view)
        initData()
    }
}
